test/unit_tests/unittest_timeout_adjust_func.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 Pilz GmbH & Co. KG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#include <chrono>
+
+#include <gtest/gtest.h>
+
+#include <psen_scan/timeout_adjust_func.h>
+
+using namespace psen_scan;
+
+namespace psen_scan_test
+{
+TEST(TimeoutAdjustFuncTests, testTimeoutIncrease)
+{
+  constexpr auto my_timeout{ std::chrono::seconds(20) };
+  constexpr auto timeout_increase{ std::chrono::seconds(17) };
+  ASSERT_EQ(my_timeout + timeout_increase, adjustTimeout(my_timeout, timeout_increase)) << "Incorrect timeout returned";
+}
+
+TEST(TimeoutAdjustFuncTests, testMaxTimeoutExceed)
+{
+  constexpr auto my_timeout{ std::chrono::seconds(75) };
+  constexpr auto timeout_increase{ std::chrono::seconds(7) };
+  constexpr auto min_allowed_timeout{ std::chrono::seconds(25) };
+  constexpr auto max_allowed_timeout{ std::chrono::seconds(77) };
+
+  ASSERT_EQ(min_allowed_timeout, adjustTimeout(my_timeout, timeout_increase, max_allowed_timeout, min_allowed_timeout))
```

Comment:
As mentioned earlier I think this should be:
```suggestion
  ASSERT_EQ(max_allowed_timeout, adjustTimeout(my_timeout, timeout_increase, max_allowed_timeout, min_allowed_timeout))
```

Reply:
Done.